core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -126,7 +126,9 @@ class KafkaApis(val requestChannel: RequestChannel,
   }
 
   private def validateForwardRequest(request: RequestChannel.Request): Unit = {
-    if (!request.context.fromPrivilegedListener ||
+    if (!config.forwardingEnabled) {
+      throw new InvalidRequestException(s"The handling of forwarding request $request is not enabled on this broker.")
```

Comment:
Can we use `closeConnections`. We do not want to even acknowledge that the api exists unless forwarding is enabled.

Reply:
It's a bit tricky to do it here since we rely on exception catching to skip all the rest of handling logic, not sure it is worth to add this special case and do `if-else` to incur a large code change.

Reply:
Perhaps we could return a boolean. I think it is important to avoid exposing this api until we're ready for it.